regression/goto-analyzer/value-set-function-pointers-simple/test.desc

```diff
@@ -6,20 +6,20 @@ main.c
 ^file main.c line 33 function main: replacing function pointer by 2 possible targets$
 ^file main.c line 41 function main: replacing function pointer by 2 possible targets$
 ^file main.c line 49 function main: replacing function pointer by 2 possible targets$
-^main::1::fun0 \(\) -> TOP$
-^main::1::fun1 \(\) -> value-set-begin: ptr ->\(f\), :value-set-end$
-^main::1::fun2_show \(\) -> value-set-begin: (TOP, )?ptr ->\([fg]\), (TOP, )?ptr ->\([fg]\), (TOP, )?:value-set-end$
-^main::1::fun3_show \(\) -> value-set-begin: ptr ->\([fg]\), ptr ->\([fg]\), :value-set-end$
-^fun_global_show \(\) -> value-set-begin: (TOP, )?ptr ->\([fg]\), (TOP, )?ptr ->\([fg]\), (TOP, )?:value-set-end$
+^main::1::fun0 \(\) -> TOP
+^main::1::fun1 \(\) -> value-set-begin: ptr ->\(f\) :value-set-end
+^main::1::fun2_show \(\) -> value-set-begin: ptr ->\(f\), ptr ->\(g\) :value-set-end
+^main::1::fun3_show \(\) -> value-set-begin: ptr ->\(f\), ptr ->\(g\) :value-set-end
+^fun_global_show \(\) -> value-set-begin: ptr ->\(f\), ptr ->\(g\) :value-set-end
 ^EXIT=0$
 ^SIGNAL=0$
 --
 ^warning: ignoring
-^main::1::fun0_show \(\) -> value-set-begin: .* :value-set-end$
-^main::1::fun1_show \(\) -> value-set-begin: .*ptr ->\(h\).* :value-set-end$
-^main::1::fun2_show \(\) -> value-set-begin: .*ptr ->\(h\).* :value-set-end$
-^main::1::fun3_show \(\) -> \{\[0\] = value-set-begin: .*ptr ->\(h\).* :value-set-end$
-^fun_global_show \(\) -> value-set-begin: .*ptr ->\(h\).* :value-set-end$
+^main::1::fun0_show \(\) -> value-set-begin: .* :value-set-end
+^main::1::fun1_show \(\) -> value-set-begin: .*ptr ->\(h\).* :value-set-end
+^main::1::fun2_show \(\) -> value-set-begin: .*ptr ->\(h\).* :value-set-end
+^main::1::fun3_show \(\) -> \{\[0\] = value-set-begin: .*ptr ->\(h\).* :value-set-end
+^fun_global_show \(\) -> value-set-begin: .*ptr ->\(h\).* :value-set-end
 --
 These TOP values in the sets shouldn't exist.
 They're caused by a quirk in the implementation, see
```

Comment:
The next line references:

https://github.com/diffblue/cbmc/issues/5307 

if most of the erronous `TOP` values are gone then maybe the comment should go too?  Also possibly the issue?

Reply:
I'm looking now at splitting value-sets of constants from value-sets-of-pointers at the moment. I'm not sure it will necessarily solve everything, but it should at least help separate concerns and highlight where the holes are.

Reply:
OK.  I am open to both split and non-split solutions, also I'm fine with it being in a subsequent PR.  However the comment about TOP should probably leave with the commit that removes them.

Reply:
Fixed